factory/ideas.php

```diff
@@ -265,9 +253,29 @@ public function get_idea($id)
 	}
 
 	/**
+	 * Returns an idea specified by its topic ID.
+	 *
+	 * @param int $id The ID of the idea to return.
+	 *
+	 * @return array The idea
+	 */
+	public function get_idea_by_topic_id($id)
```

Comment:
We're adding 2 queries per every viewtopic page here, probably we could cache at least the following one. Alternatively, cache all ideas IDs and get the one needed from it if exists.